Dashboard/app/js/lib/components/devices/AssignmentsEditView/index.jsx

```diff
@@ -0,0 +1,201 @@
+/* eslint-disable import/no-unresolved */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import FolderSurveySelectorView from 'akvo-flow/components/selectors/FolderSurveySelector';
+import FormSelectorView from 'akvo-flow/components/selectors/FormSelector';
+import DeviceGroupSelectorView from 'akvo-flow/components/selectors/DeviceSelector';
+import AssignmentDetails from './AssignmentDetails';
+
+export default class AssignmentsEditView extends React.Component {
+  state = {
+    assignmentName: this.props.inputValues.assignmentName,
+    startDate: this.props.inputValues.startDate,
+    expireDate: this.props.inputValues.toDate,
+    nameValidationMsg: '',
+  }
+
+  // lifecycle methods
+  componentDidMount() {
+    this.validateAssignment(this.props.inputValues.assignmentName);
+  }
+
+  // event handlers
+  onChangeState = (key, value) => {
+    if (key === 'assignmentName') this.validateAssignment(value);
+    this.setState({ [key]: value });
+  }
+
+  onSubmit = () => {
+    const { assignmentName, startDate, expireDate } = this.state;
+    if (this.assignmentNotComplete()) return;
+
+    this.props.actions.onSubmit({ assignmentName, startDate, expireDate });
+  }
+
+  // helpers
+  validateAssignment = (assignmentName) => {
```

Comment:
As its validating only the assignment name maybe it makes sense to specify this?

```suggestion
  validateAssignmentName = (assignmentName) => {
```